src/libs/route_utils.js

```diff
@@ -0,0 +1,30 @@
+
+export function formatDuration(sec) {
+  sec = Math.max(60, sec); // For duration < 60s, return '1 min'
+  let min = Math.round(sec / 60);
+
+  if (min < 60) {
+    return `${min}&nbsp;min`;
+  }
+
+  const hour = Math.floor(min / 60);
+  min = min - 60 * hour;
+  let ret = `${hour}&nbsp;h`;
+  if (min > 0 && hour < 10) {
+    ret += `&nbsp;${min < 10 ? '0' : ''}${min}&nbsp;min`;
+  }
+  return ret;
+}
+
+export function formatDistance(m) {
+  if (m > 99000) {
+    return `${Math.round(m / 1000)}&nbsp;km`;
+  }
+  if (m > 1000) {
```

Comment:
We had a debate about this but I still prefer using `else if` in such cases. But you can completely ignore this comment.

Reply:
Seriously, this pyramid was ridiculous…
```
if (m > 5) {
  if (m > 1000) {
    if (m > 99000) {
```

Reply:
That wasn't my point. I didn't like that pyramid either.